staging/src/k8s.io/apiserver/pkg/util/webhook/validation.go

```diff
@@ -25,6 +26,19 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
+func IsValidCABundle(caBundle []byte) bool {
+	certPool := x509.NewCertPool()
+	return certPool.AppendCertsFromPEM(caBundle)
+}
+
+func ValidateCABundle(fldPath *field.Path, caBundle []byte) field.ErrorList {
+	var allErrors field.ErrorList
+	if !IsValidCABundle(caBundle) {
+		allErrors = append(allErrors, field.Invalid(fldPath, caBundle, "Error parsing CABundle"))
```

Comment:
This error is going to be misleading if the bundle is non-empty but contains no certificates

Should this use the same client-go method that the bundle will actually be used with instead?

```go
_, err := transport.TLSConfigFor(&transport.Config{TLS: transport.TLSConfig{CAData: caBundle}})
if err != nil {
  allErrors = append(allErrors, field.Invalid(fldPath, caBundle, err.Error())
}
```

Reply:
hm I thought we could directly extract that validation via https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/transport.go#L125 and https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/transport.go#L260-L265. Is this missing the non-empty but contains no certificates case? I can change it then

Reply:
Updated.